src/averageAges.js

```diff
@@ -55,7 +65,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => onlyWithSon
+    ? person.sex === 'm' && findMother(people, person.mother)
+    : findMother(people, person.mother));
+
+  const ageDifference = children.map(person =>
+    person.born - motherAge(people, person.mother));
+  const ageDifferenceSum = ageDifference.reduce((prev, x) => prev + x);
+
+  return ageDifferenceSum / children.length;
+
+  function findMother(arr, mothersName) {
+    return arr.some(person => person.name === mothersName);
+  }
+
+  function motherAge(arr, mothersName) {
```

Comment:
```suggestion
  function getMotherAge(arr, mothersName) {
```
we usually name functions as a verb